core/src/main/java/io/wcm/testing/mock/aem/MockContentFragment_MockFragmentData.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * #%L
+ * wcm.io
+ * %%
+ * Copyright (C) 2023 wcm.io
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package io.wcm.testing.mock.aem;
+
+import java.util.Calendar;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.adobe.cq.dam.cfm.ContentFragmentException;
+import com.adobe.cq.dam.cfm.DataType;
+import com.adobe.cq.dam.cfm.FragmentData;
+
+class MockContentFragment_MockFragmentData implements FragmentData {
+
+  private Object value;
+  private String contentType;
+  private final MockContentFragment_MockDataType mockDataType;
+
+  MockContentFragment_MockFragmentData(Object value, boolean isArray) {
+    this.value = value;
+    this.mockDataType = new MockContentFragment_MockDataType(isArray);
+  }
+
+  @Override
+  public @NotNull DataType getDataType() {
+    return mockDataType;
+  }
+
+  @Override
+  public <T> @Nullable T getValue(Class<T> type) {
+    if (type.isInstance(value)) {
+      return (T)value;
+    }
+    else {
+      return null;
+    }
+  }
+
+  @Override
+  public boolean isTypeSupported(Class type) {
+    return type.isInstance(value);
```

Comment:
i think this is too simplfied as well? following https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/adobe/cq/dam/cfm/FragmentData.html#isTypeSupported(java.lang.Class) it should return true for any type supported to set as value (supported by the internal converters). it's not related to the value currently set. for sake of simplicity we could just return true here with a comment that this is simplified.